### PR TITLE
Feature/162 implement axios

### LINKED ISSUE #162: Use Axios for network requests
Reasons:
1. Support for cross browsers
2. Interceptor to set the bearer token rather than passing through state.

https://github.com/axios/axios

### PR DESCRIPTION
**Pull request by tjanowsk**
Replaced fetch with axios.

Also fixed a bug where an engagement could not be edited/removed before refreshing the screen first.

closes #162

### CODE REVIEW

File: frontend/src/services/apiService.ts [hunk 1]
```diff
@@ -1,18 +1,29 @@
...
-const apiRequest = (method: string) => async (
-  request: RequestInfo,
+const axiosInstance = axios.create({
+  baseURL: environment.baseUrl,
+  timeout: 1000,
+});
+
+const apiRequest = (method: 'GET' | 'PUT' | 'POST' | 'DELETE') => async (
+  url: string,
   token?: string,
```
Comment (connorsettle) [inline frontend/src/services/apiService.ts:16]: is it possible we can set the token via interceptor?

Comment (connorsettle) [reply frontend/src/services/apiService.ts:16]: Not sure how straightforward it will be with current auth0 implementation

Comment (tjanowsk) [reply frontend/src/services/apiService.ts:16]: I think useAuth0 has to be wrapped in a react component at the moment, I'll see if there is a workaround.

Comment (tjanowsk) [reply frontend/src/services/apiService.ts:16]: This seems like a fairly big task and more than one way of going about this so I opened #185

**Pull request merged**